CRM/MembershipExtras/Hook/Alter/CalculatedMembershipStatus.php

```diff
@@ -66,16 +66,20 @@ public function __construct() {
    *   Membership details from the calling function
    */
   public function alterMembershipStatus(&$calculatedStatus, $arguments, $membership) {
+    // If membership hasn't been created, do not alter status.

```

Comment:
A better approach rather than using an inline comment is : 

```php
$isMembershipExist = CRM_Utils_Array::value('id', $this->membership, false);
if (!$isMembershipExist) {
return;
}
```